### PR TITLE
Update Docstring for 'from_delayed' Function 

### LINKED ISSUE #8568: Document `dask.DataFrame.from_delayed` compatibility with `Future`s
`dask.DataFrame.from_delayed` works when either an iterable of `Delayed` objects or an iterable of `Future`s are passed in as input. However, the docstring for `from_delayed`

https://github.com/dask/dask/blob/358a5e367eedc8f2963651071a72f43e0ac4f887/dask/dataframe/io/io.py#L573-L576

only mentions compatibly with `Delayed` objects. We should update the `from_delayed` docstring to also mentioned `Future`s can be used

### PR DESCRIPTION
**Pull request by Kirito1397**
- [x] Closes #8568
~~- [ ] Tests added / passed~~
- [x] Passes `pre-commit run --all-files`

Updated the docstring for `from_delayed` function in `"dask/dataframe/io.io.py"` file, documenting the `dask.DataFrame.from_delayed` compatibility for both `Delayed` and  `Future` objects.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: dask/dataframe/io/io.py [hunk 1]
```diff
@@ -570,10 +570,11 @@ def from_delayed(
 
     Parameters
     ----------
-    dfs : list of Delayed
+    dfs : list of Delayed or Future
         An iterable of ``dask.delayed.Delayed`` objects, such as come from
-        ``dask.delayed`` These comprise the individual partitions of the
-        resulting dataframe.
+        ``dask.delayed`` or an iterable of ``concurrent.futures.Future`` objects,
```
Comment (pavithraes) [inline dask/dataframe/io/io.py:575]: We're talking about [Dask's implementation of `concurrent.futures`](https://docs.dask.org/en/stable/futures.html), so this should be:

"an iterable of `distributed.Future`"

Ref: https://distributed.dask.org/en/stable/api.html#future

File: dask/dataframe/io/io.py [hunk 2]
```diff
@@ -570,10 +570,11 @@ def from_delayed(
...
+        such as come from ``concurrent.futures``. These comprise the individual
```
Comment (pavithraes) [inline dask/dataframe/io/io.py:576]: And, `Future` objects are created by [all of these functions](https://distributed.dask.org/en/stable/memory.html?highlight=future%20object#creating-futures), so based on [the language used here](https://distributed.dask.org/en/stable/#architecture), maybe we can say:

"such as come from `client.submit` interface."

**Review by pavithraes — COMMENTED** [review]
@Kirito1397 Welcome to Dask and thanks for your PR!

File: dask/dataframe/io/io.py (hunk 2 above)
Comment (Kirito1397) [reply dask/dataframe/io/io.py:576]: Yes, that seems correct. I will make the change to reflect it.

File: dask/dataframe/io/io.py (hunk 1 above)
Comment (Kirito1397) [reply dask/dataframe/io/io.py:575]: Thank you! for the insight @pavithraes, but just out of curiosity isn't it a good idea to use `dask.distributed.Future` here ?

Comment (Kirito1397) [reply dask/dataframe/io/io.py:575]: Just verified now that the distributed class is in another repo. So, we should indeed use `distributed.Future`. If possible, will you be able to provide insight on it for my better understanding?

Kindly excuse me for my incompetency.

**Comment by jsignell** [comment]
ok to test

File: dask/dataframe/io/io.py (hunk 1 above)
Comment (Kirito1397) [reply dask/dataframe/io/io.py:575]: Understood @pavithraes, I will update the PR now to reflect the changes.

**Comment by Kirito1397** [comment]
Updated the Pull Request to reflect the suggested changes. Kindly review it.

**Comment by jsignell** [comment]
Thanks @Kirito1397 this is in!

**Comment by Kirito1397** [comment]
You are Welcome and Thank you @pavithraes for your assistance.

**Pull request merged**